zebra-consensus/src/block/check.rs

```diff
@@ -53,6 +58,43 @@ pub fn is_equihash_solution_valid(header: &Header) -> Result<(), equihash::Error
 /// accepted." [§7.5][7.5]
 ///
 /// [7.5]: https://zips.z.cash/protocol/protocol.pdf#blockheader
-pub fn is_time_valid_at(header: &Header, now: DateTime<Utc>) -> Result<(), BoxError> {
+pub fn time_is_valid_at(header: &Header, now: DateTime<Utc>) -> Result<(), BoxError> {
     header.is_time_valid_at(now)
 }
+
+/// [3.9]: https://zips.z.cash/protocol/protocol.pdf#subsidyconcepts
+pub fn subsidy_is_correct(network: Network, block: &Block) -> Result<(), BlockError> {
```

Comment:
I also made `subsidy_is_correct` use match in c0ffed5586782ef3c01efd25507c7e83e35570b8 in oxarbitrage#29.

I discovered a bunch of testnet bugs: the funding streams are based on the first halving, not Canopy. (They are the same height on mainnet, but different on testnet.)